apps/backend/src/libs/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,45 @@
+import { type FastifyReply, type FastifyRequest } from "fastify";
+import fp from "fastify-plugin";
+
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { token } from "~/libs/modules/token/token.js";
```

Comment:
you need to pass service through parameters to the plugin

Reply:
```
type AuthorizationPluginOptions = {
	token: Token;
	userService: UserService;
  }

const authorization = fp<AuthorizationPluginOptions>((fastify, options, done) => {
	const { token, userService } = options;
```

and then add options property to server??
`private options: Options;`
```
	private initPlugins(): void {
		const { token, userService } = this.options;

        this.app.register(authorization, { token, userService });
	}
```

Reply:
right, add it to the options